processing/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java

```diff
@@ -300,9 +308,13 @@ private static Stream<InputSplit<List<CloudObjectLocation>>> toSplitStream(
             o -> {
               try {
                 if (o.getSize() == CloudObjectSplitWidget.UNKNOWN_SIZE) {
-                  return new InputFileAttribute(splitWidget.getObjectSize(o.getLocation()));
+                  return new InputFileAttribute(
+                      inputFormat.getWeightedSize(
```

Comment:
Do we want ability to turn this weighted scaling off? im thinking we do, as if a user has specified a maxBytes config, previously and have been ingesting one of the input format types / compression that are now being scaled, they will use more tasks then they did before possibly, which is likely undesirable as it will likely be more costly for them. i think that we can turn this on / off with a new parameter for the splitHintSpec. Default can be false. Let me know what you think of this